api/v1/qdrantcluster_types.go

```diff
@@ -15,8 +15,22 @@ const (
 )
 
 const (
-	// The Annotation key to trigger a restart.
+	// RestartedAtAnnotationKey is the annotation key to trigger a restart.
+	// The annotation should be placed on the QdrantCluster instance.
+	// The value should be a [RFC3339 formatted] date.
+	// If the value is updated it will retrigger the restart.
+	// The operator will not remove the annotation once restarted.
+	// For historical reasons the key doesn't start with `operator.qdrant.com/`
 	RestartedAtAnnotationKey = "restartedAt"
+	// RecreateNodeAnnotationKey is the annotation key to recreate a certain node.
+	// The annotation should be placed on the QdrantCluster instance.
+	// Multiple annotations can be added, the operator will handle them all.
+	// The value should contain the node index to be replaced (e.g. "3").
+	// The node index should be one of the values specified in .Status.AvailableNodeIndexes.
+	// If a none existing node index is provided, the operator will remove the annotation (with a warning in the log).
```

Comment:
The operator should not remove the annotation. Kubernetes behavior is by design [level-based](https://github.com/kubernetes/community/blob/a7de9ec822008a2e5bc1a3761f8fdbd248d38dff/contributors/devel/sig-architecture/api-conventions.md#objects) and hence an observer might not see annotation creation, trying to create it then, leading potentially to a loop. 

An operator should remove annotation or revert a resource changes only as a part of resource protection.

Reply:
Removed it